File: ts/src/day1/day1_a.ts
import * as fs from "fs";
import readline from "readline";

async function countNumOfIncDepths(): Promise<number> {
	let count = 0;

	const rl = readline.createInterface({
		input: fs.createReadStream("./input"),
	});
	let prev;
	for await (const line of rl) {
		const cur = Number.parseInt(line);

		if (prev == undefined) prev = cur;

		if (cur > prev) count++;

		prev = cur;
	}

	return new Promise<number>((resolve) => resolve(count));
}

(function main() {
	countNumOfIncDepths().then(console.log);
})();

File: golang/day1/day1_a.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func countNumOfIncDepths() int {
	count := 0
	file, _ := os.Open("./input")

	scanner := bufio.NewScanner(file)

	scanner.Scan()
	prev, _ := strconv.Atoi(scanner.Text())

	for scanner.Scan() {
		cur, _ := strconv.Atoi(scanner.Text())
		if cur > prev {
			count++
		}

		prev = cur
	}

	return count
}

func main() {
	fmt.Println(countNumOfIncDepths())
}
